oracle/oracle.go

```diff
@@ -786,3 +786,92 @@ func nodePoolContainsNode(s []containerengine.Node, e string) bool {
 	}
 	return false
 }
+
+func (o *oracleOps) SetClusterVersion(version string, timeout time.Duration) error {
+	logrus.Println("Setting Cluster version to", version)
+	req := containerengine.UpdateClusterRequest{
+		ClusterId: &o.clusterID,
+		UpdateClusterDetails: containerengine.UpdateClusterDetails{
+			KubernetesVersion: &version,
+		},
+	}
+
+	resp, err := o.containerEngine.UpdateCluster(context.Background(), req)
+	if err != nil {
+		return err
+	}
+
+	return o.waitTillWorkStatusIsSucceeded(resp.OpcRequestId, resp.OpcWorkRequestId, timeout)
+}
+
+func (o *oracleOps) SetInstanceGroupVersion(instanceGroupName string, version string, timeout time.Duration) error {
+	logrus.Println("Setting Instance group version to", version)
+	//get nodepool ID from name
+	var instanceGroupID *string
+	nodePoolsReq := containerengine.ListNodePoolsRequest{CompartmentId: &o.compartmentID, Name: &instanceGroupName, ClusterId: &o.clusterID}
+	nodePools, err := o.containerEngine.ListNodePools(context.Background(), nodePoolsReq)
+	if err != nil {
+		return err
+	}
+
+	if len(nodePools.Items) == 0 {
+		return errors.New("No node pool found with name" + instanceGroupName)
+	}
+	instanceGroupID = nodePools.Items[0].Id
+
+	//update kubernetes version of nodepool
+	resp, err := o.containerEngine.UpdateNodePool(context.Background(), containerengine.UpdateNodePoolRequest{
+		NodePoolId: instanceGroupID,
+		UpdateNodePoolDetails: containerengine.UpdateNodePoolDetails{
+			KubernetesVersion: &version,
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	err = o.waitTillWorkStatusIsSucceeded(resp.OpcRequestId, resp.OpcWorkRequestId, timeout)
+	if err != nil {
+		return err
+	}
+
+	//get existing total node count of instanceGroup
```

Comment:
Put all the steps of scaling down to 0 and scaling back to original size, in separate private helper function and just call that here.

Reply:
Done